test/e2e/testing-manifests/storage-csi/update-hostpath.sh

```diff
@@ -111,31 +112,40 @@ EOF
 images=$(grep -r '^ *image:.*csi' hostpath/hostpath | sed -e 's;.*image:.*/;;' | grep -v 'node-driver-registrar' | sort -u)
 for image in $images; do
     tag=$(echo "$image" | sed -e 's/.*://')
-    path=
+    localpath=
+    remotepath=
     rbac="rbac.yaml"
     case $image in
         csi-external-*)
             # csi-external-health-monitor-agent:v0.2.0
             project=$(echo "$image" | sed -e 's/csi-\(.*\)-[^:]*:.*/\1/')
-            path=$(echo "$image" | sed -e 's/csi-\([^:]*\):.*/\1/')
+            localpath=$(echo "$image" | sed -e 's/csi-\([^:]*\):.*/\1/')
+            remotepath=$localpath
             ;;
         *)
             project=$(echo "$image" | sed -e 's/:.*//' -e 's/^csi/external/')
             case $project in
                 external-snapshotter)
                     # Another special case...
-                    path="csi-snapshotter"
+                    localpath="csi-snapshotter"
+                    remotepath=$localpath
                     rbac="rbac-csi-snapshotter.yaml"
                     ;;
+                external-snapshot-metadata)
+                    # Another special case
+                    remotepath=".."
+                    localpath=""
+                    rbac="snapshot-metadata-cluster-role.yaml"
+                    ;;
             esac
             ;;
     esac
-    download "$project" "$path" "$tag" "$rbac"
+    download "$project" "$localpath" "$remotepath" "$tag" "$rbac"
 done
 
 # Update the mock driver manifests, too.
 grep -r image: hostpath/hostpath/csi-hostpath-plugin.yaml | while read -r image; do
     version=$(echo "$image" | sed -e 's/.*:\(.*\)/\1/')
     image=$(echo "$image" | sed -e 's/.*image: \([^:]*\).*/\1/')
-    sed -i '' -e "s;$image:.*;$image:$version;" mock/*.yaml
```

Comment:
This change might cause the script to fail on macOS. sed doesn't support `-i` without a value on macOS. Please keep it as is.

Reply:
Suggestion Changes:

```
if sed --version >/dev/null 2>&1; then
  # GNU sed (Linux/Fedora)
  sed -i -e "s;$image:.*;$image:$version;" mock/*.yaml
else
  # BSD sed (macOS)
  sed -i '' -e "s;$image:.*;$image:$version;" mock/*.yaml
fi
```

Reply:
Updated